src/charm.py

```diff
@@ -183,7 +175,7 @@ def __init__(self, *args, **kwargs):
 
         self.framework.observe(self.on.leader_elected, self._setup_replica_data)
         self.framework.observe(self.on.start, self._on_start)
-        self.framework.observe(self.on.uploads_storage_attached, self._on_storage_attached)
+        self.framework.observe(self.on.uploads_storage_attached, self._reconciliation)
```

Comment:
Is this event triggered before the config change? If so, we can probably remove this listener given that the callback is the same
`self.framework.observe(self.on.config_changed, self._reconciliation)`

Reply:
I am not sure if the firing order of `*-storage-attached` event is guaranteed, it might fired after the `config-changed`. I just add a callback here to make sure the charm doesn't stuck waiting for storage.